wizard-frontend/src/components/UserDataContext.tsx

```diff
@@ -0,0 +1,45 @@
+import { LayerFileConfig } from "@/mirrord-schema";
+import React, { useState } from "react";
+
+export const UserDataContext = React.createContext<boolean | undefined>(
+  undefined
+);
+
+export interface ConfigData {
+  config: LayerFileConfig;
+  setConfig: (config: LayerFileConfig) => void;
+}
+
+export const DefaultConfig: LayerFileConfig = {
+  feature: {
+    network: {
+      incoming: {
+        mode: "mirror",
+      },
+      outgoing: true,
+    },
+    fs: "read",
+    env: true,
+  },
+};
+
+export const ConfigDataContextProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [config, setConfig] = useState<LayerFileConfig>(DefaultConfig);
+  const setConfigWithLog = (config: LayerFileConfig) => {
+    console.log("Setting config:", config);
+    setConfig(config);
+  };
+  return (
+    <ConfigDataContext.Provider value={{ config, setConfig: setConfigWithLog }}>
+      {children}
+    </ConfigDataContext.Provider>
+  );
+};
+
+export const ConfigDataContext = React.createContext<ConfigData | undefined>(
+  undefined
+);
```

Comment:
Why `ConfigData | undefined`, our config supports default values for everything right? So I would assume that we can initialize a config here and just go modifying the default values?